app/controllers/searches_controller.rb

```diff
@@ -8,6 +8,7 @@ def show
     @error_msg, @gems = Rubygem.search(params[:query], es: es_enabled?, page: @page)
     limit_total_entries if @gems.total_entries > MAX_PAGE * Rubygem.per_page
 
+    @all_gems =  @gems.reject { |gem| gem == @gems.first }
```

Comment:
Isn't this equivalent to `@gems.uniq!`? You can skip extra variable `@all_gems` too.